### PR TITLE
handle hardhat config js paths

### PR DESCRIPTION
**Pull request by jnsdls**

### CODE REVIEW

File: src/core/builder/hardhat.ts [hunk 1]
```diff
@@ -14,17 +16,43 @@ export class HardhatBuilder implements IBuilder {
...
   }> {
+    // TODO this will work for .js but we also need to handle .ts cases
+    let hardhatConfig: HardhatUserConfig = {
+      paths: {
+        artifacts: "./artifacts",
+        sources: "./contracts",
+      },
+    };
+    try {
+      const readHardHatConfig: HardhatUserConfig = require(join(
```
Comment (joaquim-verges) [inline src/core/builder/hardhat.ts]: I assume this is how hardhat loads its own config file? What do they do for .ts ?

File: src/core/builder/hardhat.ts [hunk 2]
```diff
@@ -14,17 +16,43 @@ export class HardhatBuilder implements IBuilder {
...
     logger.debug("stdout from hardhat:", stdout.toString());
 
-    const artifacts = `${options.projectPath}/artifacts/contracts`;
+    const artifactsPath = join(
+      options.projectPath,
+      hardhatConfig.paths?.artifacts || "./artifacts"
+    );
+    const contractsPath = join(
+      artifactsPath,
+      hardhatConfig.paths?.sources || "./contracts"
```
Comment (joaquim-verges) [inline src/core/builder/hardhat.ts:61]: But annoying to repeat these defaults twice no? Default hardhat config can just be empty object i guess

**Review by joaquim-verges — APPROVED** [review]
Just some Q's, feel free to merge and keep at it

File: src/core/builder/hardhat.ts (hunk 2 above)
Comment (jnsdls) [reply src/core/builder/hardhat.ts:61]: yeh exactly

**Pull request merged**